tools/ld-analyse/CMakeLists.txt

```diff
@@ -0,0 +1,246 @@
+find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core Gui Widgets HINTS ${QT_LIBRARY_DIR})
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Gui Widgets)
+
+# Name all the files
+add_executable(ld-analyse MACOSX_BUNDLE
+    blacksnranalysisdialog.cpp blacksnranalysisdialog.h blacksnranalysisdialog.ui
+    busydialog.cpp busydialog.h busydialog.ui
+    closedcaptionsdialog.cpp closedcaptionsdialog.h closedcaptionsdialog.ui
+    main.cpp
+    mainwindow.cpp mainwindow.h mainwindow.ui
+    oscilloscopedialog.cpp oscilloscopedialog.h oscilloscopedialog.ui
+    vectorscopedialog.cpp vectorscopedialog.h vectorscopedialog.ui
+    aboutdialog.cpp aboutdialog.h aboutdialog.ui
+    videoparametersdialog.cpp videoparametersdialog.h videoparametersdialog.ui
+    chromadecoderconfigdialog.cpp chromadecoderconfigdialog.h chromadecoderconfigdialog.ui
+    tbcsource.cpp tbcsource.h
+    vbidialog.cpp vbidialog.h vbidialog.ui
+    configuration.cpp configuration.h
+    dropoutanalysisdialog.cpp dropoutanalysisdialog.h dropoutanalysisdialog.ui
+    ../ld-chroma-decoder/palcolour.cpp ../ld-chroma-decoder/palcolour.h
+    ../ld-chroma-decoder/comb.cpp ../ld-chroma-decoder/comb.h
+    ../ld-chroma-decoder/componentframe.cpp ../ld-chroma-decoder/componentframe.h
+    ../ld-chroma-decoder/outputwriter.cpp ../ld-chroma-decoder/outputwriter.h
+    ../ld-chroma-decoder/transformpal.cpp ../ld-chroma-decoder/transformpal.h
+    ../ld-chroma-decoder/transformpal2d.cpp ../ld-chroma-decoder/transformpal2d.h
+    ../ld-chroma-decoder/transformpal3d.cpp ../ld-chroma-decoder/transformpal3d.h
+    ../ld-chroma-decoder/framecanvas.cpp ../ld-chroma-decoder/framecanvas.h
+    ../ld-chroma-decoder/sourcefield.cpp ../ld-chroma-decoder/sourcefield.h
+    ../library/filter/firfilter.h
+    ../library/tbc/dropouts.cpp ../library/tbc/dropouts.h
+    ../library/tbc/filters.cpp ../library/tbc/filters.h
+    ../library/tbc/jsonio.cpp ../library/tbc/jsonio.h
+    ../library/tbc/lddecodemetadata.cpp ../library/tbc/lddecodemetadata.h
+    ../library/tbc/linenumber.h
+    ../library/tbc/logging.cpp ../library/tbc/logging.h
+    ../library/tbc/sourcevideo.cpp ../library/tbc/sourcevideo.h
+    ../library/tbc/vbidecoder.cpp ../library/tbc/vbidecoder.h
+    visibledropoutanalysisdialog.cpp visibledropoutanalysisdialog.h visibledropoutanalysisdialog.ui
+    whitesnranalysisdialog.cpp whitesnranalysisdialog.h whitesnranalysisdialog.ui
+)
+
+# Get the current working branch
+execute_process(
+	COMMAND git rev-parse --abbrev-ref HEAD
+	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+	OUTPUT_VARIABLE GIT_BRANCH
+	OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Get the latest abbreviated commit hash of the working branch
+execute_process(
+	COMMAND git log -1 --format=%h
+	WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+	OUTPUT_VARIABLE GIT_COMMIT_HASH
+	OUTPUT_STRIP_TRAILING_WHITESPACE
+)
+
+# Pass it on to the source code
+add_definitions(-DAPP_BRANCH="\""+${GIT_BRANCH}+"\"" -DAPP_COMMIT="\""+${GIT_COMMIT_HASH}+"\"")
```

Comment:
This is deprecated - use [add_compile_definitions](https://cmake.org/cmake/help/latest/command/add_compile_definitions.html) which will make the quoting a bit less horrible.